File: ocExample/ocExample/ViewController.m
//
//  ViewController.m
//  ocExample
//
//  Created by yang chuang on 2021/5/2.
//

#import "ViewController.h"
#import <NPSMeter-Swift.h>
#import <Masonry/Masonry.h>
#import <SVProgressHUD/SVProgressHUD.h>
#import <IQKeyboardManagerSwift-Swift.h>

@interface ViewController ()<UITextFieldDelegate>

@property (nonatomic, strong) UITextField *textField;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    IQKeyboardManager.shared.enable = YES;
    
    UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"demo"]];
    [self.view addSubview:imageView];
    [imageView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.equalTo(@0);
    }];
    
    UIImageView *logo = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"logo"]];
    [self.view addSubview:logo];
    [logo mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(@24);
        make.top.equalTo(@55);
        make.width.equalTo(@100);
        make.height.equalTo(@(100*logo.image.size.height/logo.image.size.width));
    }];
    
    
    UIButton *startButton = [UIButton buttonWithType:UIButtonTypeCustom];
    [startButton setTitle:@"显示问卷" forState:UIControlStateNormal];
    startButton.backgroundColor = [UIColor colorWithRed:0 green:0.54 blue:1 alpha:1];
    [startButton setTitleColor:UIColor.whiteColor forState:UIControlStateNormal];
    [self.view addSubview:startButton];
    [startButton addTarget:self action:@selector(startButtonAction) forControlEvents:UIControlEventTouchUpInside];
    [startButton mas_makeConstraints:^(MASConstraintMaker *make) {
        make.bottom.equalTo(@(-64));
        make.left.equalTo(@24);
        make.right.equalTo(@(-24));
        make.height.equalTo(@48);
    }];
    startButton.layer.masksToBounds = YES;
    startButton.layer.cornerRadius = 3;
    
    self.textField = [[UITextField alloc] init];
    self.textField.text = @"9e637a54b426cc17";
    self.textField.textAlignment = NSTextAlignmentCenter;
    self.textField.backgroundColor = UIColor.whiteColor;
    [self.view addSubview:self.textField];
    [self.textField mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.height.right.equalTo(startButton);
        make.bottom.equalTo(startButton.mas_top).offset(-16);
    }];
    self.textField.layer.masksToBounds = YES;
    self.textField.layer.cornerRadius = 3;
    self.textField.delegate = self;
    self.textField.textColor = [UIColor colorWithRed:0.165 green:0.192 blue:0.333 alpha:1];
    self.textField.attributedPlaceholder = [[NSAttributedString alloc] initWithString:@"请输入问卷id" attributes:@{NSForegroundColorAttributeName:[UIColor colorWithRed:0.580 green:0.596 blue:0.667 alpha:1]}];
    [self downConfig];
    
    UILabel *label = [[UILabel alloc] init];
    label.text = @"新一代应用内调研工具";
    label.font = [UIFont boldSystemFontOfSize:28];
    [self.view addSubview:label];
    [label mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.textField);
        make.bottom.equalTo(self.textField.mas_top).offset(-32);
    }];
}

- (void)downConfig {
    
    [NPSMeter downConfigWithId:self.textField.text downResult:^(BOOL success, NSString * _Nullable error) {
        if (success) {
            [SVProgressHUD showSuccessWithStatus:@"下载配置成功"];
        } else {
            UIAlertController *alertView = [UIAlertController alertControllerWithTitle:error message:nil preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *retryAction = [UIAlertAction actionWithTitle:@"重试" style:UIAlertActionStyleDestructive handler:^(UIAlertAction * _Nonnull action) {
                [self downConfig];
            }];
            UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"取消" style:UIAlertActionStyleCancel handler:nil];
            [alertView addAction:retryAction];
            [alertView addAction:cancelAction];
            [self presentViewController:alertView animated:YES completion:nil];
        }
    }];
}

- (void)startButtonAction {
    [NPSMeter showWithId:self.textField.text showResult:^(BOOL success, NSString * _Nullable error) {
        if (error) {
            [SVProgressHUD showErrorWithStatus:error];
        }
    }];
}

- (void)textFieldDidEndEditing:(UITextField *)textField {
    [self downConfig];
}

@end

File: Example/NPSMeter/ViewController.swift
//
//  ViewController.swift
//  NPSMeter
//
//  Created by mumenma on 05/02/2021.
//  Copyright (c) 2021 mumenma. All rights reserved.
//

import UIKit
import NPSMeter
import RxSwift
import RxCocoa
import PKHUD
import IQKeyboardManagerSwift

class ViewController: UIViewController,UITextFieldDelegate {
    
    var disposeBag = DisposeBag()
    let textField = UITextField()

    override func viewDidLoad() {
        super.viewDidLoad()
        
        IQKeyboardManager.shared.enable = true
        
        let imageView = UIImageView(image: UIImage(named: "demo"))
        self.view.addSubview(imageView)
        imageView.snp.makeConstraints { make in
            make.edges.equalToSuperview()
        }
        
        
        let logo = UIImageView(image: UIImage(named: "logo"))
        self.view.addSubview(logo)
        logo.snp.makeConstraints { make in
            make.left.equalTo(24)
            make.top.equalTo(55)
            make.width.equalTo(100)
            make.height.equalTo(100*(logo.image?.size.height)!/(logo.image?.size.width)!)
        }
                
        let startButton = UIButton(type: UIButton.ButtonType.custom)
        startButton.setTitle("显示问卷", for: UIControl.State.normal)
        startButton.backgroundColor = UIColor(red: 0, green: 0.54, blue: 1, alpha: 1)
        startButton.setTitleColor(UIColor.white, for: UIControl.State.normal)
        self.view.addSubview(startButton)
        startButton.rx.tap.subscribe{ (_) in
            NPSMeter.show(id:self.textField.text!, showResult:{ (success,error) in
                if error != nil {
                    HUD.flash(.label(error), delay: 2.0, completion: { (_) in
                                
                    })
                }
            });
        }.disposed(by: disposeBag)
        startButton.snp.makeConstraints { make in
            make.bottom.equalTo(-64);
            make.left.equalTo(24);
            make.right.equalTo(-24);
            make.height.equalTo(48);
        }
        startButton.layer.masksToBounds = true
        startButton.layer.cornerRadius = 3
        
        self.view.addSubview(self.textField)
        self.textField.text = "9e637a54b426cc17"
        self.textField.textAlignment = NSTextAlignment.center
        self.textField.backgroundColor = UIColor.white
        self.textField.snp.makeConstraints { make in
            make.left.height.right.equalTo(startButton)
            make.bottom.equalTo(startButton.snp.top).offset(-16)
        }
        self.textField.layer.masksToBounds = true
        self.textField.layer.cornerRadius = 3
        self.textField.delegate = self
        self.textField.textColor = UIColor(red: 0.165, green: 0.192, blue: 0.333, alpha: 1)
        self.textField.attributedPlaceholder = NSAttributedString(string: "请输入问卷id", attributes:[NSAttributedString.Key.foregroundColor:UIColor(red: 0.580, green: 0.596, blue: 0.667, alpha: 1)])
        self.downConfig()
        
        let label = UILabel()
        label.text = "新一代应用内调研工具"
        label.font = UIFont.boldSystemFont(ofSize: 28)
        self.view.addSubview(label)
        label.snp.makeConstraints { make in
            make.left.equalTo(self.textField)
            make.bottom.equalTo(self.textField.snp.top).offset(-32)
        }
    }
    
    func downConfig() {
        NPSMeter.downConfig(id:self.textField.text!){ (success,error) in
            if (success) {
                HUD.flash(.label("下载配置成功"), delay: 2.0, completion: { (_) in
                            
                })
            } else {
                let alertView = UIAlertController(title:error, message: nil, preferredStyle: UIAlertController.Style.alert)
                let alertAction1 = UIAlertAction(title: "重试",style: .destructive, handler: {
                        action in
                    self.downConfig()
                })
                let alertAction2 = UIAlertAction(title: "取消", style: .cancel)
                alertView.addAction(alertAction1)
                alertView.addAction(alertAction2)
                self.present(alertView, animated: true, completion: nil);
            }
        }
    }
    
    
    func textFieldDidEndEditing(_ textField: UITextField) {
        self.downConfig()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        IQKeyboardManager.shared.keyboardDistanceFromTextField = 80
    }

}

